### PR TITLE
Npc/battle tests

### PR DESCRIPTION
**Pull request by northofsophia**
We have added numerous tests for various npc_battle integration such as set functions, get functions, change functions, and functions for looting dead bodies.

### CODE REVIEW

**Review by amymorrill — CHANGES_REQUESTED** [review]
Looks good! I just saw a small bug in my tests (which I'll fix), and I also wonder if the test for a dead npc with multiple items should check that the correct number of items were transferred.

File: tests/npc/test_npc_battle.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    add_item_to_npc(npc, test_item2);
+    add_item_to_npc(npc, test_item3);
+
+    cr_assert_not_null(npc->npc_battle, "add_battle_to_npc() failed");
+    cr_assert_not_null(npc->inventory, "add_item_to_npc() failed to add item");
+
+    int rc = transfer_all_npc_items(npc, room);
+
+    cr_assert_eq(rc, SUCCESS, "transfer_all_npc_items() failed");
+    cr_assert_not_null(room->items,
```
[the remark this answers is not in the document]
Comment (northofsophia) [reply tests/npc/test_npc_battle.c]: I have added an additional check that sees if the correct items were transferred.

**Review by amymorrill — APPROVED** [review]

File: tests/npc/test_npc.c [hunk 2]
```diff
@@ -34,6 +35,53 @@ npc_mov_t *generate_test_npc_mov()
     npc_mov = npc_mov_new(NPC_MOV_DEFINITE, test_room);
 }
 
+/* Creates example stats. Taken from test_battle_ai.c */
+stat_t* create_enemy_stats()
```
Comment (lilyehsani) [inline tests/npc/test_npc.c:39]: Move * over to be `stat_t *create_enemy_stats()` etc throughout this file

**Review by lilyehsani — CHANGES_REQUESTED** [review]
Looks really good! Added one comment, a style thing that needs to be changed in various places.

File: tests/npc/test_npc.c [hunk 3]
```diff
@@ -34,11 +35,58 @@ npc_mov_t *generate_test_npc_mov()
...
+
+     move->attack = attack;
+     move->damage = damage;
+     move->defense = defense;
+
+     return move;
+}
+
+/* Creates example moves. Taken from test_battle_ai.c */
+move_t* create_enemy_moves()
```
Comment (lilyehsani) [inline tests/npc/test_npc.c:73]: move *

**Review by lilyehsani — CHANGES_REQUESTED** [review]
Missed 1 spot

File: tests/npc/test_npc.c (hunk 3 above)
Comment (northofsophia) [reply tests/npc/test_npc.c]: I have fixed this style change

**Review by lilyehsani — APPROVED** [review]
LGTM! Approved.

**Comment by amymorrill** [comment]
The tests that we wrote for the npc_battle related functions that we created this quarter have now been merged into dev. We created a new file, tests/npc/test_npc_battle.c, that contains tests for functions in the npc_battle module and added to the file tests/npc/test_npc.c. This pr only included unit testing and did not include any integration testing with the battle module, which will have to be added by next year's class.

**Comment by lilyehsani** [comment]
Issue Score: ✔️+

Comments:
Great work overall! I have two comments:
-  The title of a PR has to convey what the PR will be producing. In the future, try using simple titles with a "Verb-Object" structure that clearly states what you will be doing in the PR (e.g., “Design and implement a ‘circle’ module”, “Research libraries for drawing simple geometric shapes”, etc.)
- The summary for this PR is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/policies.html#tasks-issues for examples).

**Pull request merged**